Modules/Cart/Http/Services/CartService.php

```diff
@@ -261,108 +277,123 @@ public function fetch($requestData): array
+                        $cartArr[$brandK]['products'][$prdK]['product_qty'] = $prdV['quantity'];
+                        $cartArr[$brandK]['products'][$prdK]['style_name'] = $prdV['style_name'];
+                        $cartArr[$brandK]['products'][$prdK]['style_group_name'] = $prdV['style_group_name'];
+                        $cartArr[$brandK]['products'][$prdK]['type'] = $prdV['type'];
+                        $cartArr[$brandK]['products'][$prdK]['product_image'] = $product->featured_image != '' ? $product->featured_image : asset('public/img/logo-image.png');
                     }
                 }
             }
         }
 
         $data = array(
             'cart_count' => $cartCount,
-            'cart_arr' => $cart_arr,
+            'cart_arr' => $cartArr,
         );
 
         $response = ['res' => true, 'msg' => '', 'data' => $data];
+
         return ($response);
     }
 
-
     /**
-     * @param array $cartData
+     * User can delete product from his cart
+     *
+     * @param $id
+     * @param $user_id
      * @return array
      */
-
-    public function update(array $cartData): array
+    public function delete($id, $user_id): array
     {
-        //update Program
-        $id = $cartData['id'];
-        $name = $cartData['name'];
-        $country = $cartData['country'];
-        $street = $cartData['street'];
-        $suite = $cartData['suite'];
-        $state = $cartData['state'];
-        $town = $cartData['town'];
-        $zip = $cartData['zip'];
-        $phoneCode = $cartData['phoneCode'];
-        $data = array(
-            'name' => $name,
-            'country' => $country,
-            'street' => $street,
-            'suite' => $suite,
-            'state' => $state,
-            'town' => $town,
-            'zip' => $zip,
-            'phoneCode' => $phoneCode
-        );
-        Cart::where('id', $id)->update($data);
+        $cart = Cart::where('id', $id)->where('user_id', $user_id)->first();
+        if (empty($cart)) {
+            return [
+                'res' => false,
+                'msg' => 'Error please try again',
+                'data' => ""
+            ];
+        }
+        $cart->delete();
 
-        $response = [
+        return [
             'res' => true,
-            'msg' => 'Updated Successfully',
-            'data' => ''
+            'msg' => 'Cart successfully removed',
+            'data' => ""
         ];
-
-        return $response;
     }
 
     /**
-     * @param $requestData
+     * Update existing cart
+     *
+     * @param $request
      * @return array
      */
-    public function delete($id, $user_id): array
+    public function update($request): array
     {
-        $cart = Cart::where('id', $id)->where('user_id', $user_id)->first();
-        if (empty($cart)) {
+        $carts = $request->cart;
+        $error = 0;
+        foreach ($carts as $v) {
+            $cart = Cart::where('id', $v['id'])->where('user_id', $v['user_id'])->first();
```

Comment:
Should be $cart = Cart::where('id', $v['id'])->first();
do not use user_id from front end, use auth()->user()->id.

Reply:
Done